autoload/ale/references.vim

```diff
@@ -51,18 +53,44 @@ function! ale#references#HandleLSPResponse(conn_id, response) abort
 
         if type(l:result) is v:t_list
             for l:response_item in l:result
-                call add(l:item_list, {
-                \ 'filename': ale#path#FromURI(l:response_item.uri),
-                \ 'line': l:response_item.range.start.line + 1,
-                \ 'column': l:response_item.range.start.character + 1,
-                \})
+                let l:col = l:response_item.range.start.character + 1
+                let l:lnum = l:response_item.range.start.line + 1
+                let l:filename = ale#path#FromURI(l:response_item.uri)
+
+                if g:ale_use_quickfix_findref
+                    let l:cwd = getcwd()
+                    call add(l:item_list, {
+                    \ 'filename': substitute(l:filename, l:cwd . '/', "", ""),
+                    \ 'lnum': l:lnum,
+                    \ 'col': l:col,
+                    \ 'text': system('sed "' . l:lnum . 'q;d" ' . l:filename),
```

Comment:
You'll have to do something other than a `system()` call here. This will slow down Vim a lot, and it won't work on Windows.